clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -808,7 +808,8 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
                 }
                 log.debug("Subscribed to topic(s): {}", Utils.join(topics, ", "));
                 this.subscriptions.subscribe(topics, listener);
-                metadata.setTopics(subscriptions.groupSubscription());
+                this.metadata.needMetadataForAllTopics(true);
```

Comment:
@hachikuji If I'm not mistaken you are suggesting to use this for metadata update:
```
                if (!this.metadata.topics().containsAll(topics))
                    this.metadata.requestUpdate();
```

The issue is if we just reduce the subscribed topics (e.g. from `[foo, bar, quux]` to `[foo, quux]`) with this code the metadata won't be updated until the next metadata refresh timeout (which is 5 minutes by default). Unless we use a check that verifies the two sets (metadata topics, and newly subscribed topics) are not the same?